src/ncp/rcp_host.cpp

```diff
@@ -344,6 +345,18 @@ void RcpHost::HandleStateChanged(otChangedFlags aFlags)
     }
 
     mThreadHelper->StateChangedCallback(aFlags);
+
+    if ((aFlags & OT_CHANGED_THREAD_ROLE) && IsAttached() && mJoinReceiver != nullptr)
+    {
+        otOperationalDatasetTlvs datasetTlvs;
+
+        GetDatasetActiveTlvs(datasetTlvs);
+        if (datasetTlvs.mLength != 0)
```

Comment:
What's the reason to do this check for non-empty dataset (in addition to `IsAttahced()` check)?

Reply:
This code is converted from the Android implementation. But thinking on this, this is not needed.